src/context.rs

```diff
@@ -54,6 +54,14 @@ impl Context {
     pub fn cancel(&self) {
         self.shared.cancel();
     }
+
+    pub fn cancel_rx(&self) -> Receiver<()> {
+        self.cancel_rx.clone()
```

Comment:
I think you might need to increment num_chans here. But also not sure how you can decrement it if the channel gets dropped. These channels aren't really intended to be apart if the public api, why do you need them?

Reply:
@notoriaga I was trying to use it here:
https://github.com/swift-nav/swift-toolbox/pull/487